esx_service/vmdk_ops.py

```diff
@@ -844,24 +848,48 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     except ValidationError as ex:
         return err(str(ex))
 
-    if not datastore:
-        datastore = default_datastore
-
     if datastore and not vmdk_utils.validate_datastore(datastore):
         return err("Invalid datastore '%s'.\n" \
                    "Known datastores: %s.\n" \
                    "Default datastore: %s" \
                    % (datastore, ", ".join(get_datastore_names_list()), default_datastore))
 
```

Comment:
Again, datastore list should be whats available for the tenant. Is it possible for a VM in a tenant to be able to create volumes on any datastore all the time? Or is it possible only if ALL_DS is set for the tenant.